### DISCUSSION TITLE
Accidentally re-starring a repo shows me as a now-current stargazer

### DISCUSSION THREAD

**Discussion by RocketRobz** [community/community#66303] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Feature Area

Issues

### Body

I wanted to view the stargazer list of a repository, so I clicked the `Starred` button, not knowing it would unstar me from the repo, so I clicked it again to fix the mistake.
After properly opening the stargazer list, the mistake is not properly fixed, as it now shows me as a recent/current stargazer, rather than somewhere in the list when I starred it for the first time.

Please fix it so that if a user accidentally unstars a repo, it would not place them as a recent/current stargazer.
How that could work is for the user to wait sometime around either 10 seconds or 1 minute. If the user waits longer, they get displayed as a recent or a current stargazer.

I've opened this discussion, because I've recently done this with this repo: https://github.com/Gericom/GBARunner3
The first time I starred this repo was sometime in 7/29/2023. By re-starring the repo, it would give some users the wrong idea that I found out about the repo pretty late, which is not true.

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @RocketRobz ! 

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Projects and Issues category is solely related to conversations around the GitHub products Issues and Projects. I've gone ahead and moved it for you. Good luck!

**Comment by justinclift** [comment]
Yeah.  This has been a problem since GitHub changed the meaning of the buttons, so you have to click something in a different place to view who starred a repo.

That being said, it's still possible to get it wrong regardless of button placement (mis-click, lack of coffee, etc), which effectively "resets" when you starred the repo.

It'd be nice to have this "fixed", such that re-starring a repository uses the original datestamp from the first time the user starred it.
